instrumentation/opentelemetry/init.go

```diff
@@ -243,14 +251,15 @@ func InitWithSpanProcessorWrapper(cfg *config.AgentConfig, wrapper SpanProcessor
 	metricsShutdownFn := initializeMetrics(cfg, versionInfoAttrs)
 
 	exporterFactory = makeExporterFactory(cfg)
+	configFactory = makeConfigFactory(cfg)
 
 	exporter, err := exporterFactory()
 	if err != nil {
 		log.Fatal(err)
 	}
 
 	sp := modbsp.CreateBatchSpanProcessor(
-		cfg.GetTelemetry() != nil && cfg.GetTelemetry().GetMetricsEnabled().GetValue(), // metrics enabled
```

Comment:
Just to understand this more, why was this conditional to run BatchSpanProcesor until now? What I am trying to understand is what would happen if customBsp is false and metrics are enabled? Will non-modified batchSpanProcessor work?

Reply:
The only reason for the custom BatchSpanProcessor(BSP) is to be able to collect span metrics - received, dropped and unsampled - as they hit the exporter. So if metrics is disabled, there's no need for it.
As if this PR if custom BSP is false and metrics are enabled, we use the stock BSP. I think this behavior will remain even in the recover from panic PR. And I'm thinking I will remove the metrics enabled requirement in that PR so that we can recover even if metrics are disabled and panic happens when exporting the traces.